src/register.ts

```diff
@@ -0,0 +1,48 @@
+import axios from 'axios'
+import { registerMethod } from 'did-resolver'
+
+const DOC_PATH = '/.well-known/did.json'
+
+export default function register() {
+  async function resolve(
+    did: string,
+    parsed: ParsedDID
+  ): Promise<DIDDoc | null> {
+    const url: string = `https://${parsed.id}${DOC_PATH}`
+    try {
+      const res = await axios.get(url)
+      // even if the requested url is https, an http response
+      // will be 200 so check the request agent protocol
+      // (would parsing res.request.res.responseUrl be better?)
+      const isHttp = res.request.agent.protocol.trim() === 'http'
```

Comment:
Not sure how this compares to methods of detecting that a response was sent over HTTP instead of HTTPS. Alternatives are welcome.

Reply:
I think in this case it would always be https since you're creating the URL yourself on line 11

Reply:
That's what I thought initially as well, but when I tested locally using [badssl.com](https://github.com/chromium/badssl.com), doing `axios.get('https://http.badssl.test')` returned a 200 response over `http`. Didn't find any explanations online about this behavior, but I can try testing with other http libs to address (your comment below)[https://github.com/uport-project/https-did-resolver/pull/1#discussion_r219369445] as well

Reply:
@pelle It was doing an automatic redirect from the https URL to the http one. Handled this scenario using `xmlhttprequest` instead of `axios` in this [PR](https://github.com/uport-project/https-did-resolver/pull/2) and only accepting 200 response and not automatically following 302 redirect. Is this an acceptable solution or do we need more logic around handling non-error status codes?